seckey.h

```diff
@@ -80,16 +80,16 @@ class VariableRounds
 	//! \throws InvalidRounds if the number of rounds are invalid
 	inline void ThrowIfInvalidRounds(int rounds, const Algorithm *alg)
 	{
-#if defined(__BORLANDC__)
-		if (rounds < MIN_ROUNDS || rounds > MAX_ROUNDS)
-			throw InvalidRounds(alg ? alg->AlgorithmName() : std::string("VariableRounds"), rounds);
-#elif (M==INT_MAX)	// Coverity and result_independent_of_operands
-		if (rounds < MIN_ROUNDS)
-			throw InvalidRounds(alg ? alg->AlgorithmName() : "VariableRounds", rounds);	
-#else
-		if (rounds < MIN_ROUNDS || rounds > MAX_ROUNDS)
-			throw InvalidRounds(alg ? alg->AlgorithmName() : "VariableRounds", rounds);
-#endif
+		if (M == INT_MAX) // Coverity and result_independent_of_operands
+		{
+			if (rounds < MIN_ROUNDS)
+				throw InvalidRounds(alg ? alg->AlgorithmName() : std::string("VariableRounds"), rounds);
+		}
+		else
+		{
+			if (rounds < MIN_ROUNDS || rounds > MAX_ROUNDS)
+				throw InvalidRounds(alg ? alg->AlgorithmName() : std::string("VariableRounds"), rounds);
+		}
 	}
 

```

Comment:
@noloader I have compiled Crypto++ with the modified version of _seckey.h_ using C++Builder XE6 and XE8 and there is no problems. I think that Blacktempel is right when he says that the problem occurs with `M == INT_MAX` preprocessor.